addons/sale_stock/models/stock.py

```diff
@@ -51,6 +51,25 @@ def _assign_picking_post_process(self, new=False):
                     values={'self': picking_id, 'origin': sale_order_id},
                     subtype_id=self.env.ref('mail.mt_note').id)
 
+    def _generate_sale_order_lines(self):
+        sale_order_lines_vals = []
+        for move in self:
+            product = move.product_id
+            so_line = {
+                'move_ids': [(6, 0, move.ids)],
+                'name': product.display_name,
+                'order_id': move.picking_id.sale_id.id,
+                'product_id': product.id,
+                'product_uom_qty': move.product_uom_qty,
+                'price_unit': product.list_price if product.invoice_policy == 'delivery' else 0,
```

Comment:
You should probably check the price on:
- 1. Other existing line.
- 2. Vendor pricelist
- 3. list_price.